### PR TITLE
treewide (darwin): fix or enable darwin build for many packages (5)

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
See #114976 for details.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 115974` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>criterion</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>clucene_core</li>
    <li>gtkdatabox</li>
    <li>guile-opengl</li>
    <li>htmlcxx</li>
    <li>kissfft</li>
    <li>levmar</li>
    <li>libcec_platform</li>
    <li>libcli</li>
    <li>libcredis</li>
    <li>libcsptr</li>
    <li>liblcf</li>
    <li>libnatpmp</li>
    <li>libnest2d</li>
    <li>liboauth</li>
    <li>sword</li>
  </ul>
</details>

File: pkgs/development/libraries/kissfft/default.nix [hunk 1]
```diff
@@ -30,6 +30,11 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace "gcc" "${stdenv.cc.targetPrefix}cc" \
+      --replace "ar" "${stdenv.cc.targetPrefix}ar"
```
Comment (r-burns) [inline pkgs/development/libraries/kissfft/default.nix:36]: Alternatively, as of v131.1.0 these aren't hardcoded.

https://github.com/mborgerding/kissfft/commit/fcf9fa9a47c4c0fec252bbb3951ce044ca8dc129

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115974` at bfa48bf3 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>swift-im</li>
  </ul>
</details>
<details>
  <summary>12 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/b601410f29f5d10cb6649f56e0f3eb45">ardour_5</a></li></li>
    <li><a href="https://gist.github.com/32ceaa631eb418eeaf30340b16bd6bc8">htmlcxx</a></li></li>
    <li><a href="https://gist.github.com/72ad606c3f98b887d0a7dae8a70149d1">kissfft</a></li></li>
    <li><a href="https://gist.github.com/b91b54400350cd740267b1f335771a8b">levmar</a></li></li>
    <li><a href="https://gist.github.com/bfa6842abcc6b79dd2a0c16330d47681">lgogdownloader</a></li></li>
    <li><a href="https://gist.github.com/2479f696e3ecf8c9bc6004c4c2012fed">libcli</a></li></li>
    <li><a href="https://gist.github.com/057977c82669bf89764b0c850f7457fe">libcsptr</a></li></li>
    <li><a href="https://gist.github.com/81484262196e55aa6d32ce061421fdfd">libnatpmp</a></li></li>
    <li><a href="https://gist.github.com/8118e071addd20eefe7f4de0f080818d">meshlab</a></li></li>
    <li><a href="https://gist.github.com/bec9247179e3ddf4c4df92ba6d7566d2">netsniff-ng</a></li></li>
    <li><a href="https://gist.github.com/bffb3aa1542cc7bb992c09dbae81d6d5">qm-dsp</a></li></li>
    <li><a href="https://gist.github.com/76ef695ee8097e7bae5a9896026c9c7e">torrential</a></li></li>
  </ul>
</details>
<details>
  <summary>8 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/htmlcxx/default.nix:15:3:
```
   |
15 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/levmar/default.nix:23:3:
```
   |
23 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/libcli/default.nix:25:3:
```
   |
25 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/libraries/htmlcxx/default.nix:13:15:
```
   |
13 |   patches = [ ./ptrdiff.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/libraries/libcli/default.nix:15:7:
```
   |
15 |     [ (fetchurl {
   |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/libraries/levmar/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/libraries/levmar/default.nix:11:3:
```
   |
11 |   patchPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl2` is a deprecated license, check if project uses `lgpl2Plus` or `lgpl2Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/htmlcxx/default.nix:18:5:
```
   |
18 |     license = licenses.lgpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/htmlcxx/default.nix [hunk 2]
```diff
@@ -9,12 +9,13 @@ stdenv.mkDerivation rec {
     sha256 = "1hgmyiad3qgbpf2dvv2jygzj6jpz4dl3n8ds4nql68a4l9g2nm07";
   };
 
+  buildInputs = [ libiconv ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/htmlcxx/default.nix:12]: Is this only required for darwin?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115974` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>clucene_core (clucene_core_1)</li>
<li>gtkdatabox</li>
<li>guile-opengl</li>
<li>htmlcxx</li>
<li>kissfft</li>
<li>levmar</li>
<li>libcec_platform</li>
<li>libcli</li>
<li>libcredis</li>
<li>libcsptr</li>
<li>liblcf</li>
<li>libnatpmp</li>
<li>libnest2d</li>
<li>liboauth</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>ardour_5</li>
<li>htmlcxx</li>
<li>kissfft</li>
<li>levmar</li>
<li>lgogdownloader</li>
<li>libcli</li>
<li>libnatpmp</li>
<li>meshlab</li>
<li>netsniff-ng</li>
<li>qm-dsp</li>
<li>swift-im</li>
<li>torrential</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ardour_5:
<details>


warning: attribute-typo
A likely typo in `wafConfigureFlags` argument was found, did you mean `configureFlags`?

Near pkgs/applications/audio/ardour/5.nix:115:3:

```
    |
115 |   wafConfigureFlags = [
    |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/ardour/5.nix:158:5:

```
    |
158 |     license = licenses.gpl2;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unused-argument
Unused argument: `glibc`.
Near pkgs/applications/audio/ardour/5.nix:11:3:

```
   |
11 | , glibc
   |   ^
```
</details>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 115974` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>clucene_core (clucene_core_1)</li>
    <li>gtkdatabox</li>
    <li>guile-opengl</li>
    <li>htmlcxx</li>
    <li>kissfft</li>
    <li>levmar</li>
    <li>libcec_platform</li>
    <li>libcli</li>
    <li>libcredis</li>
    <li>libcsptr</li>
    <li>liblcf</li>
    <li>libnatpmp</li>
    <li>libnest2d</li>
    <li>liboauth</li>
  </ul>
</details>

File: pkgs/development/libraries/htmlcxx/default.nix (hunk 2 above)
Comment (siraben) [reply pkgs/development/libraries/htmlcxx/default.nix:12]: Yes but it's a no-op for linux since it's included with glibc.

**Comment by siraben** [comment]
Adding another batch of fixes.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 115974` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>criterion</li>
  </ul>
</details>
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>clucene_core</li>
    <li>coan</li>
    <li>gtkdatabox</li>
    <li>guile-opengl</li>
    <li>htmlcxx</li>
    <li>kissfft</li>
    <li>levmar</li>
    <li>libcec_platform</li>
    <li>libcli</li>
    <li>libcredis</li>
    <li>libcsptr</li>
    <li>liblcf</li>
    <li>lrdf (librdf)</li>
    <li>libnatpmp</li>
    <li>libnest2d</li>
    <li>liboauth</li>
    <li>libosip</li>
    <li>libxmi</li>
    <li>libxmp</li>
    <li>lightning</li>
    <li>loudmouth</li>
    <li>metal</li>
    <li>minixml</li>
    <li>msgpuck</li>
    <li>perl530Packages.AlienTidyp</li>
    <li>perl530Packages.HTMLTidy</li>
    <li>perl532Packages.AlienTidyp</li>
    <li>perl532Packages.HTMLTidy</li>
    <li>pwndbg</li>
    <li>pxlib</li>
    <li>safefile</li>
    <li>shhmsg</li>
    <li>shhopt</li>
    <li>spglib</li>
    <li>spice-protocol (spice_protocol)</li>
    <li>sword</li>
    <li>tidyp</li>
    <li>vassal</li>
  </ul>
</details>

File: pkgs/development/libraries/kissfft/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/development/libraries/kissfft/default.nix:36]: Can this be fetched with fetchpatch?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>ardour_5</li>
<li>htmlcxx</li>
<li>kissfft</li>
<li>levmar</li>
<li>lgogdownloader</li>
<li>libcli</li>
<li>libnatpmp</li>
<li>meshlab</li>
<li>netsniff-ng</li>
<li>qm-dsp</li>
<li>shhmsg</li>
<li>shhopt</li>
<li>snake4</li>
<li>swift-im</li>
<li>torrential</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ardour_5:
<details>


warning: attribute-typo
A likely typo in `wafConfigureFlags` argument was found, did you mean `configureFlags`?

Near pkgs/applications/audio/ardour/5.nix:115:3:

```
    |
115 |   wafConfigureFlags = [
    |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/ardour/5.nix:158:5:

```
    |
158 |     license = licenses.gpl2;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unused-argument
Unused argument: `glibc`.
Near pkgs/applications/audio/ardour/5.nix:11:3:

```
   |
11 | , glibc
   |   ^
```
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115974` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
33 packages built:
</summary>
<ul>
<li>clucene_core (clucene_core_1)</li>
<li>coan</li>
<li>gtkdatabox</li>
<li>guile-opengl</li>
<li>htmlcxx</li>
<li>kissfft</li>
<li>levmar</li>
<li>libcec_platform</li>
<li>libcli</li>
<li>libcredis</li>
<li>libcsptr</li>
<li>liblcf</li>
<li>libnatpmp</li>
<li>libnest2d</li>
<li>liboauth</li>
<li>libosip</li>
<li>libxmi</li>
<li>libxmp</li>
<li>lightning</li>
<li>loudmouth</li>
<li>lrdf</li>
<li>metal</li>
<li>minixml</li>
<li>msgpuck</li>
<li>pwndbg</li>
<li>pxlib</li>
<li>safefile</li>
<li>shhmsg</li>
<li>shhopt</li>
<li>spglib</li>
<li>spice-protocol</li>
<li>tidyp</li>
<li>vassal</li>
</ul>
</details>

**Pull request merged**